library/src/main/java/com/google/android/exoplayer/metadata/Id3Parser.java

```diff
@@ -67,6 +67,39 @@ public boolean canParse(String mimeType) {
         String value = new String(frame, valueStartIndex, valueEndIndex - valueStartIndex,
             charset);
         metadata.put(TxxxMetadata.TYPE, new TxxxMetadata(description, value));
+      } else if (frameId0 == 'P' && frameId1 == 'R' && frameId2 == 'I' && frameId3 == 'V') {
+        // Check frame ID == PRIV
+        byte[] frame = new byte[frameSize];
+        id3Data.readBytes(frame, 0, frameSize);
+
+        int firstZeroIndex = indexOf(frame, 0, (byte) 0);
+        String owner = new String(frame, 0, firstZeroIndex);
+        byte[] privateData = new byte[frameSize - firstZeroIndex - 1];
+        System.arraycopy(frame, firstZeroIndex + 1, privateData, 0, frameSize - firstZeroIndex - 1);
+        metadata.put(PrivMetadata.TYPE, new PrivMetadata(owner, privateData));
+      } else if (frameId0 == 'G' && frameId1 == 'E' && frameId2 == 'O' && frameId3 == 'B') {
+        // Check frame ID == GEOB
+        int encoding = id3Data.readUnsignedByte();
+        String charset = getCharsetName(encoding);
+        byte[] frame = new byte[frameSize - 1];
+        id3Data.readBytes(frame, 0, frameSize - 1);
+
+        int firstZeroIndex = indexOf(frame, 0, (byte) 0);
+        String mimeType = new String(frame, 0, firstZeroIndex);
+        int filenameStartIndex = firstZeroIndex + 1;
+        int filenameEndIndex = indexOf(frame, filenameStartIndex, (byte) 0);
+        String filename = new String(frame, filenameStartIndex,
+                                     filenameEndIndex - filenameStartIndex, charset);
+        int descriptionStartIndex = filenameEndIndex + 1;
```

Comment:
Sorry, not sure about this change. In some cases filename and/or description fields are empty fields and then just a byte 0 is found in their locations (from ID3 format spec: "The first two
   strings may be omitted, leaving only their terminations"). This means that, for example, in case filename is fullfilled (with value = "any", for example) but description is empty I will found the following in frame byte array:
// x == filenameStart
frame[x] = 'a'
frame[x+1] = 'n'
frame[x+2] = 'y'
frame[x+3] = 0 // End of filename field
frame[x+4] = '0' // End of description field, because it is empty
frame[x+5] = // Start of objectData field

This scenario is happening, for example, in the description field of GEOB frames included in Apple test stream. They are always set as empty.

In those cases if I use indexOfNot(frame, filenameEndIndex, (byte) 0); then I would skip the byte 0 indicating description field is empty and will take objectData field as the description one.

Am I wrong? Am I missing anything?